src/services/serverRequest.ts

```diff
@@ -0,0 +1,35 @@
+import { FormTypes } from '@oneblink/types'
+import { generateHeaders } from './fetch'
+
+function getOBApi(apiId: string, apiEnv: string): string {
+  if (apiEnv.toLowerCase() === 'prod') {
+    return apiId
+  }
+  const arr = apiId.split('.')
+  arr[0] += `-${apiEnv}`
+  return arr.join('.')
+}
+
+export default async function serverRequest(
+  endpoint: FormTypes.Form['serverValidation'],
+  payload: Record<string, unknown>,
+): Promise<{ response: Response; url: string } | void> {
+  if (!endpoint) return
+
+  const url =
+    endpoint.type === 'CALLBACK'
+      ? endpoint.configuration.url
+      : `https://${getOBApi(
+          endpoint.configuration.apiId,
+          endpoint.configuration.apiEnvironment,
+        )}${endpoint.configuration.apiEnvironmentRoute}`
+
+  const headers = await generateHeaders()
+  const response = await fetch(url, {
+    method: 'POST',
+    headers,
+    body: JSON.stringify(payload),
+  })
```

Comment:
Is it possible to replace this with the `fetch` service in apps?

Reply:
I don't believe we should use the pre-existing fetch.ts in apps for Server Validation + External Id, I think it's fine if we move the new fetch code into the same file but I think the pre-existing code would be better kept handling our API calls as opposed to working also with APIs made by customers/us that we may not always be sure what the data returned will be
Happy to discuss further as I'm happy to be swayed to using it